### PR TITLE
Added page about Pod Security Admissions

### PR DESCRIPTION
**Pull request by akosma**
As the title says.

## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `how-to`, `tutorial`, `reference`, `explanation`, `cicd`, `dependency`
      as they show up in the changelog
- [ ] Link this PR to related issues if applicable.

### CODE REVIEW

File: docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+|*Baseline*
+|Minimally restrictive policy which prevents known privilege escalations.
+Allows the default (minimally specified) Pod configuration.
+
+|*Restricted*
+|Heavily restricted policy, following current Pod hardening best practices.
+|===
+
+The most straightforward way to ensure your workloads can work on a restricted namespace, particularly in 4.12, is by labeling the namespaces where they should run by enforcing the restricted policy and verifying if they're admitted and are successfully running.
```
Comment (simu) [inline docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc:43]: Note: regular APPUiO Cloud users currently don't have sufficient permissions to add the relevant labels:

```
simon@phoenix:~ [appuio-lab] $ kubectl label ns sg-test pod-security.kubernetes.io/restricted=enforce
Error from server: admission webhook "validate.kyverno.svc-fail" denied the request: 

resource Namespace//sg-test was blocked due to the following policies

validate-namespace-metadata:
  validate-labels: |-
    validation failure: The following labels can be modified:
        appuio.io/organization, custom.appuio.io/*, kubernetes.io/metadata.name, network-policies.syn.tools/no-defaults, network-policies.syn.tools/purge-defaults.
    labels given:
        {"appuio.io/organization":"vshn","kubernetes.io/metadata.name":"sg-test","pod-security.kubernetes.io/restricted":"enforce"}.
    labels before modification:
        {"appuio.io/organization":"vshn","kubernetes.io/metadata.name":"sg-test"}.
```

However, it should be fairly straightforward to allow users to set the `pod-security.kubernetes.io/restricted` label on their namespaces.

File: docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc [hunk 2]
```diff
@@ -0,0 +1,103 @@
+= Pod Security Admissions
+
+Kubernetes 1.24 removes the Pod Security Policy feature in favor of Pod Security Admission. This document explains the impact on {product} users.
+
+== Introduction
+
+Up until OpenShift 4.10, OpenShift only had Security Context Constraints (SCCs.)
```
Comment (simu) [inline docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc:7]: I assume the following is rejected by Vale, becuase to me having the period inside the parentheses looks just wrong:

```suggestion
Up until OpenShift 4.10, OpenShift only had Security Context Constraints (SCCs).
```

**Review by simu — APPROVED** [review]
LGTM overall.

The periods inside the parentheses noting the abbreviations in the introduction look absolutely broken in my understanding on how a sentence including parentheses is supposed to look, but IIRC that's Vale being Vale.

File: docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+. By the global Pod Security Admission configuration, all namespaces will run in the “restricted” enforcement mode. This Pod Security Admission (PSA) level aligns with the restricted-v2 SCC, so if your workload is categorized into restricted-v2, it will be able to run in a namespace that enforces the PSA as restricted.
+. In cases where the workload has access to higher SCC privileges (via its Service Account,) OpenShift will label namespaces, via the new label synchronization component, to synchronize the PSA enforcement level in order to keep workloads running. 
+
+The primary recommendation at this point is to ensure any workload has the security context properly set.
+Therefore, the best path is to ensure the workloads can run with restricted permissions.
+When this is done, your product will run on restricted namespaces on vanilla Kubernetes and will have the best chance of running properly on any other Kubernetes distribution with any vendor within restricted namespaces.
+The application will also have access to the restricted-v2 on OpenShift and won't be rejected by cluster administrators trying to enforce restrictions.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc:102]: Maybe wrap in an admonition.

File: docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc [hunk 4]
```diff
@@ -0,0 +1,103 @@
+= Pod Security Admissions
+
+Kubernetes 1.24 removes the Pod Security Policy feature in favor of Pod Security Admission. This document explains the impact on {product} users.
+
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc:4]: Preferably, add a tl;dr

1. Check PSA warnings
2. Set security context accordingly in your workload resources.

**Review by corvus-ch — CHANGES_REQUESTED** [review]
* Please make it clear, that APPUiO Cloud users can not set the PSA related namespace labels.
* Please fix the interpunctuation.
* Please at a call to action to the beginning of the document.

File: docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc (hunk 1 above)
Comment (akosma) [reply docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc:43]: So how can then users make sure that their workloads can work on a restricted namespace?

Comment (simu) [reply docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc:50]: IMO we could just allow them to set the `pod-security.kubernetes.io/restricted` label (this should never enable a user to grant themselves extra privileges, afaict), but maybe @corvus-ch has a different/better idea?

File: docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+|*Baseline*
+|Minimally restrictive policy which prevents known privilege escalations.
+Allows the default (minimally specified) Pod configuration.
+
+|*Restricted*
+|Heavily restricted policy, following current Pod hardening best practices.
+|===
+
+The most straightforward way to ensure your workloads can work on a restricted namespace, particularly in 4.12, is by labeling the namespaces where they should run by enforcing the restricted policy and verifying if they're admitted and are successfully running.
+You may also check whether they're assigned the restricted-v2 SCC in 4.11. We recommend validating and ensuring the desired behavior by doing functionality end-to-end testing.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc:51]:
```suggestion
On {product}, the default is `baseline` and can not be changed.
If you need more security and do want to set `restricted`, please visit http://roadmap.appuio.cloud/ and let us know. 
```

File: docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc (hunk 1 above)
Comment (corvus-ch) [reply docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc:50]: Let us, for now, not suggest changing the labels. I created APUB-347 to make that possible. As part of that, we will then update this section again.

**Review by corvus-ch — CHANGES_REQUESTED** [review]
I suggest to just not mention setting namespace labels for now. We will revisit this once we have done the implementation to make that possible.

File: docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc (hunk 1 above)
Comment (akosma) [reply docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc:50]: So we leave this section as it is now?

**Comment by akosma** [comment]
I had forgotten to add this new page to the documentation, fixed.

**Review by corvus-ch — APPROVED** [review]

**Pull request merged**